ADD: extract local archives

nalind opened the pull request:
When ADDing files from the build context that look like archives, the expected behavior is that we'll extract their contents in the destination location.  Fixes #99.

TomSweeneyRedHat commented on dockerclient/testdata/add/archived.txt:
```diff
@@ -0,0 +1 @@
+help, i've been archived
```
I've heard being archived is very painful....
;^)

TomSweeneyRedHat commented on dockerclient/client.go:
```diff
@@ -923,6 +923,28 @@ func (e *ClientExecutor) Archive(fromFS bool, src, dst string, allowDownload boo
+			}
+			return nil, false, false, nil
+		}
+		filtered, err := transformArchive(f, true, transform)
```
might be end of day brain, but where does `transformArchive` live?

nalind replied:
It's in [archive.go](https://github.com/openshift/imagebuilder/blob/master/dockerclient/archive.go#L234).  This part of imagebuilder is the direct inspiration for similar functionality [in buildah](https://github.com/containers/buildah/blob/master/digester.go#L90).

rhatdan commented:
LGTM

nalind commented:
Realized I'd changed the behavior for COPY, too.  Added a check on `allowDownload`, and added a conformance test that looks just like the new one we were already adding, but using COPY instead of ADD, so that we verify that we don't extract archives for COPY.

TomSweeneyRedHat commented:
LGTM

rhatdan commented:
/lgtm
